### DISCUSSION TITLE
Mechanism on how tailwindCSS postcss integration works in Angular

### DISCUSSION THREAD

**Discussion by classicmike** [tailwindlabs/tailwindcss#10295] [Help] [answered]
Hi all,

I know this question wouldn't explicitly be tailwind css specific but I'm going to ask anyways just so I can get an insight and be pointed in the right direction for further reading.

I'm trying to install a PostCSS plugin into our Angular project and have been tweaking around some custom webpack configuration to no avail. From what I've heard Angular doesn't really make it much easy to add your own config to your needs.

I've come across a tailwind guide on installing tailwind on angular which seemed so straightforward: https://tailwindcss.com/docs/guides/angular. 

So question is: **How does the tailwind integration work behind the scenes in angular that make it so seamless? Can you point to an area in the angular/tailwind code base that allows the integrations to happen please?**

Again hopefully pointing me to the right direction and thanks in advance.

**Comment by adamwathan** [comment, marked as answer]
Hey! I think this is the code in Angular itself that streamlines things, by checking for the presence of a Tailwind config file and automatically including Tailwind as a PostCSS plugin if it's found:

https://github.com/angular/angular-cli/blob/27b22b02dccef2e2d4f2f1f03910f4b70c676f84/packages/angular_devkit/build_angular/src/webpack/configs/styles.ts#L85-L105

**Reply by classicmike** [reply]
Hey @adamwathan,
Thanks for the answer. I'll look at the source code to read it in more detail.
